File: BackEnd/docs/apidocs/admin-core.m
#import "admin-core.h"
#ifndef DEF_ADMIN_CORENS0UiConfiguration_M
#define DEF_ADMIN_CORENS0UiConfiguration_M

/**
 * (no documentation provided)
 */
@implementation ADMIN_CORENS0UiConfiguration

- (void) dealloc
{
  [super dealloc];
}
@end /* implementation ADMIN_CORENS0UiConfiguration */

/**
 * Internal, private interface for JAXB reading and writing.
 */
@interface ADMIN_CORENS0UiConfiguration (JAXB) <JAXBReading, JAXBWriting, JAXBType>

@end /*interface ADMIN_CORENS0UiConfiguration (JAXB)*/

/**
 * Internal, private implementation for JAXB reading and writing.
 */
@implementation ADMIN_CORENS0UiConfiguration (JAXB)

/**
 * Read an instance of ADMIN_CORENS0UiConfiguration from an XML reader.
 *
 * @param reader The reader.
 * @return An instance of ADMIN_CORENS0UiConfiguration defined by the XML reader.
 */
+ (id<JAXBType>) readXMLType: (xmlTextReaderPtr) reader
{
  ADMIN_CORENS0UiConfiguration *_aDMIN_CORENS0UiConfiguration = [[ADMIN_CORENS0UiConfiguration alloc] init];
  NS_DURING
  {
    [_aDMIN_CORENS0UiConfiguration initWithReader: reader];
  }
  NS_HANDLER
  {
    _aDMIN_CORENS0UiConfiguration = nil;
    [localException raise];
  }
  NS_ENDHANDLER

  [_aDMIN_CORENS0UiConfiguration autorelease];
  return _aDMIN_CORENS0UiConfiguration;
}

/**
 * Initialize this instance of ADMIN_CORENS0UiConfiguration according to
 * the XML being read from the reader.
 *
 * @param reader The reader.
 */
- (id) initWithReader: (xmlTextReaderPtr) reader
{
  return [super initWithReader: reader];
}

/**
 * Write the XML for this instance of ADMIN_CORENS0UiConfiguration to the writer.
 * Note that since we're only writing the XML type,
 * No start/end element will be written.
 *
 * @param reader The reader.
 */
- (void) writeXMLType: (xmlTextWriterPtr) writer
{
  [super writeXMLType:writer];
}

//documentation inherited.
- (BOOL) readJAXBAttribute: (xmlTextReaderPtr) reader
{
  void *_child_accessor;

  if ([super readJAXBAttribute: reader]) {
    return YES;
  }

  return NO;
}

//documentation inherited.
- (BOOL) readJAXBValue: (xmlTextReaderPtr) reader
{
  return [super readJAXBValue: reader];
}

//documentation inherited.
- (BOOL) readJAXBChildElement: (xmlTextReaderPtr) reader
{
  id __child;
  void *_child_accessor;
  int status, depth;

  if ([super readJAXBChildElement: reader]) {
    return YES;
  }

  return NO;
}

//documentation inherited.
- (int) readUnknownJAXBChildElement: (xmlTextReaderPtr) reader
{
  return [super readUnknownJAXBChildElement: reader];
}

//documentation inherited.
- (void) readUnknownJAXBAttribute: (xmlTextReaderPtr) reader
{
  [super readUnknownJAXBAttribute: reader];
}

//documentation inherited.
- (void) writeJAXBAttributes: (xmlTextWriterPtr) writer
{
  int status;

  [super writeJAXBAttributes: writer];

}

//documentation inherited.
- (void) writeJAXBValue: (xmlTextWriterPtr) writer
{
  [super writeJAXBValue: writer];
}

/**
 * Method for writing the child elements.
 *
 * @param writer The writer.
 */
- (void) writeJAXBChildElements: (xmlTextWriterPtr) writer
{
  int status;
  id __item;
  id __item_copy;
  NSEnumerator *__enumerator;

  [super writeJAXBChildElements: writer];

}
@end /* implementation ADMIN_CORENS0UiConfiguration (JAXB) */

#endif /* DEF_ADMIN_CORENS0UiConfiguration_M */
#ifndef DEF_ADMIN_CORENS0SwaggerResource_M
#define DEF_ADMIN_CORENS0SwaggerResource_M

/**
 * (no documentation provided)
 */
@implementation ADMIN_CORENS0SwaggerResource

/**
 * (no documentation provided)
 */
- (NSString *) swaggerVersion
{
  return _swaggerVersion;
}

/**
 * (no documentation provided)
 */
- (void) setSwaggerVersion: (NSString *) newSwaggerVersion
{
  [newSwaggerVersion retain];
  [_swaggerVersion release];
  _swaggerVersion = newSwaggerVersion;
}

/**
 * (no documentation provided)
 */
- (NSString *) location
{
  return _location;
}

/**
 * (no documentation provided)
 */
- (void) setLocation: (NSString *) newLocation
{
  [newLocation retain];
  [_location release];
  _location = newLocation;
}

/**
 * (no documentation provided)
 */
- (NSString *) name
{
  return _name;
}

/**
 * (no documentation provided)
 */
- (void) setName: (NSString *) newName
{
  [newName retain];
  [_name release];
  _name = newName;
}

- (void) dealloc
{
  [self setSwaggerVersion: nil];
  [self setLocation: nil];
  [self setName: nil];
  [super dealloc];
}
@end /* implementation ADMIN_CORENS0SwaggerResource */

/**
 * Internal, private interface for JAXB reading and writing.
 */
@interface ADMIN_CORENS0SwaggerResource (JAXB) <JAXBReading, JAXBWriting, JAXBType>

@end /*interface ADMIN_CORENS0SwaggerResource (JAXB)*/

/**
 * Internal, private implementation for JAXB reading and writing.
 */
@implementation ADMIN_CORENS0SwaggerResource (JAXB)

/**
 * Read an instance of ADMIN_CORENS0SwaggerResource from an XML reader.
 *
 * @param reader The reader.
 * @return An instance of ADMIN_CORENS0SwaggerResource defined by the XML reader.
 */
+ (id<JAXBType>) readXMLType: (xmlTextReaderPtr) reader
{
  ADMIN_CORENS0SwaggerResource *_aDMIN_CORENS0SwaggerResource = [[ADMIN_CORENS0SwaggerResource alloc] init];
  NS_DURING
  {
    [_aDMIN_CORENS0SwaggerResource initWithReader: reader];
  }
  NS_HANDLER
  {
    _aDMIN_CORENS0SwaggerResource = nil;
    [localException raise];
  }
  NS_ENDHANDLER

  [_aDMIN_CORENS0SwaggerResource autorelease];
  return _aDMIN_CORENS0SwaggerResource;
}

/**
 * Initialize this instance of ADMIN_CORENS0SwaggerResource according to
 * the XML being read from the reader.
 *
 * @param reader The reader.
 */
- (id) initWithReader: (xmlTextReaderPtr) reader
{
  return [super initWithReader: reader];
}

/**
 * Write the XML for this instance of ADMIN_CORENS0SwaggerResource to the writer.
 * Note that since we're only writing the XML type,
 * No start/end element will be written.
 *
 * @param reader The reader.
 */
- (void) writeXMLType: (xmlTextWriterPtr) writer
{
  [super writeXMLType:writer];
}

//documentation inherited.
- (BOOL) readJAXBAttribute: (xmlTextReaderPtr) reader
{
  void *_child_accessor;

  if ([super readJAXBAttribute: reader]) {
    return YES;
  }

  return NO;
}

//documentation inherited.
- (BOOL) readJAXBValue: (xmlTextReaderPtr) reader
{
  return [super readJAXBValue: reader];
}

//documentation inherited.
- (BOOL) readJAXBChildElement: (xmlTextReaderPtr) reader
{
  id __child;
  void *_child_accessor;
  int status, depth;

  if ([super readJAXBChildElement: reader]) {
    return YES;
  }
  if (xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT
    && xmlStrcmp(BAD_CAST "swaggerVersion", xmlTextReaderConstLocalName(reader)) == 0
    && xmlTextReaderConstNamespaceUri(reader) == NULL) {

#if DEBUG_ENUNCIATE > 1
    NSLog(@"Attempting to read choice {}swaggerVersion of type {http://www.w3.org/2001/XMLSchema}string.");
#endif
    __child = [NSString readXMLType: reader];
#if DEBUG_ENUNCIATE > 1
    NSLog(@"successfully read choice {}swaggerVersion of type {http://www.w3.org/2001/XMLSchema}string.");
#endif

    [self setSwaggerVersion: __child];
    return YES;
  } //end "if choice"

  if (xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT
    && xmlStrcmp(BAD_CAST "location", xmlTextReaderConstLocalName(reader)) == 0
    && xmlTextReaderConstNamespaceUri(reader) == NULL) {

#if DEBUG_ENUNCIATE > 1
    NSLog(@"Attempting to read choice {}location of type {http://www.w3.org/2001/XMLSchema}string.");
#endif
    __child = [NSString readXMLType: reader];
#if DEBUG_ENUNCIATE > 1
    NSLog(@"successfully read choice {}location of type {http://www.w3.org/2001/XMLSchema}string.");
#endif

    [self setLocation: __child];
    return YES;
  } //end "if choice"

  if (xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT
    && xmlStrcmp(BAD_CAST "name", xmlTextReaderConstLocalName(reader)) == 0
    && xmlTextReaderConstNamespaceUri(reader) == NULL) {

#if DEBUG_ENUNCIATE > 1
    NSLog(@"Attempting to read choice {}name of type {http://www.w3.org/2001/XMLSchema}string.");
#endif
    __child = [NSString readXMLType: reader];
#if DEBUG_ENUNCIATE > 1
    NSLog(@"successfully read choice {}name of type {http://www.w3.org/2001/XMLSchema}string.");
#endif

    [self setName: __child];
    return YES;
  } //end "if choice"


  return NO;
}

//documentation inherited.
- (int) readUnknownJAXBChildElement: (xmlTextReaderPtr) reader
{
  return [super readUnknownJAXBChildElement: reader];
}

//documentation inherited.
- (void) readUnknownJAXBAttribute: (xmlTextReaderPtr) reader
{
  [super readUnknownJAXBAttribute: reader];
}

//documentation inherited.
- (void) writeJAXBAttributes: (xmlTextWriterPtr) writer
{
  int status;

  [super writeJAXBAttributes: writer];

}

//documentation inherited.
- (void) writeJAXBValue: (xmlTextWriterPtr) writer
{
  [super writeJAXBValue: writer];
}

/**
 * Method for writing the child elements.
 *
 * @param writer The writer.
 */
- (void) writeJAXBChildElements: (xmlTextWriterPtr) writer
{
  int status;
  id __item;
  id __item_copy;
  NSEnumerator *__enumerator;

  [super writeJAXBChildElements: writer];

  if ([self swaggerVersion]) {
    status = xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "swaggerVersion", NULL);
    if (status < 0) {
      [NSException raise: @"XMLWriteError"
                   format: @"Error writing start child element {}swaggerVersion."];
    }

#if DEBUG_ENUNCIATE > 1
    NSLog(@"writing element {}swaggerVersion...");
#endif
    [[self swaggerVersion] writeXMLType: writer];
#if DEBUG_ENUNCIATE > 1
    NSLog(@"successfully wrote element {}swaggerVersion...");
#endif

    status = xmlTextWriterEndElement(writer);
    if (status < 0) {
      [NSException raise: @"XMLWriteError"
                   format: @"Error writing end child element {}swaggerVersion."];
    }
  }
  if ([self location]) {
    status = xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "location", NULL);
    if (status < 0) {
      [NSException raise: @"XMLWriteError"
                   format: @"Error writing start child element {}location."];
    }

#if DEBUG_ENUNCIATE > 1
    NSLog(@"writing element {}location...");
#endif
    [[self location] writeXMLType: writer];
#if DEBUG_ENUNCIATE > 1
    NSLog(@"successfully wrote element {}location...");
#endif

    status = xmlTextWriterEndElement(writer);
    if (status < 0) {
      [NSException raise: @"XMLWriteError"
                   format: @"Error writing end child element {}location."];
    }
  }
  if ([self name]) {
    status = xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "name", NULL);
    if (status < 0) {
      [NSException raise: @"XMLWriteError"
                   format: @"Error writing start child element {}name."];
    }

#if DEBUG_ENUNCIATE > 1
    NSLog(@"writing element {}name...");
#endif
    [[self name] writeXMLType: writer];
#if DEBUG_ENUNCIATE > 1
    NSLog(@"successfully wrote element {}name...");
#endif

    status = xmlTextWriterEndElement(writer);
    if (status < 0) {
      [NSException raise: @"XMLWriteError"
                   format: @"Error writing end child element {}name."];
    }
  }
}
@end /* implementation ADMIN_CORENS0SwaggerResource (JAXB) */

#endif /* DEF_ADMIN_CORENS0SwaggerResource_M */
#ifndef DEF_ADMIN_CORENS0SecurityConfiguration_M
#define DEF_ADMIN_CORENS0SecurityConfiguration_M

/**
 * (no documentation provided)
 */
@implementation ADMIN_CORENS0SecurityConfiguration

- (void) dealloc
{
  [super dealloc];
}
@end /* implementation ADMIN_CORENS0SecurityConfiguration */

/**
 * Internal, private interface for JAXB reading and writing.
 */
@interface ADMIN_CORENS0SecurityConfiguration (JAXB) <JAXBReading, JAXBWriting, JAXBType>

@end /*interface ADMIN_CORENS0SecurityConfiguration (JAXB)*/

/**
 * Internal, private implementation for JAXB reading and writing.
 */
@implementation ADMIN_CORENS0SecurityConfiguration (JAXB)

/**
 * Read an instance of ADMIN_CORENS0SecurityConfiguration from an XML reader.
 *
 * @param reader The reader.
 * @return An instance of ADMIN_CORENS0SecurityConfiguration defined by the XML reader.
 */
+ (id<JAXBType>) readXMLType: (xmlTextReaderPtr) reader
{
  ADMIN_CORENS0SecurityConfiguration *_aDMIN_CORENS0SecurityConfiguration = [[ADMIN_CORENS0SecurityConfiguration alloc] init];
  NS_DURING
  {
    [_aDMIN_CORENS0SecurityConfiguration initWithReader: reader];
  }
  NS_HANDLER
  {
    _aDMIN_CORENS0SecurityConfiguration = nil;
    [localException raise];
  }
  NS_ENDHANDLER

  [_aDMIN_CORENS0SecurityConfiguration autorelease];
  return _aDMIN_CORENS0SecurityConfiguration;
}

/**
 * Initialize this instance of ADMIN_CORENS0SecurityConfiguration according to
 * the XML being read from the reader.
 *
 * @param reader The reader.
 */
- (id) initWithReader: (xmlTextReaderPtr) reader
{
  return [super initWithReader: reader];
}

/**
 * Write the XML for this instance of ADMIN_CORENS0SecurityConfiguration to the writer.
 * Note that since we're only writing the XML type,
 * No start/end element will be written.
 *
 * @param reader The reader.
 */
- (void) writeXMLType: (xmlTextWriterPtr) writer
{
  [super writeXMLType:writer];
}

//documentation inherited.
- (BOOL) readJAXBAttribute: (xmlTextReaderPtr) reader
{
  void *_child_accessor;

  if ([super readJAXBAttribute: reader]) {
    return YES;
  }

  return NO;
}

//documentation inherited.
- (BOOL) readJAXBValue: (xmlTextReaderPtr) reader
{
  return [super readJAXBValue: reader];
}

//documentation inherited.
- (BOOL) readJAXBChildElement: (xmlTextReaderPtr) reader
{
  id __child;
  void *_child_accessor;
  int status, depth;

  if ([super readJAXBChildElement: reader]) {
    return YES;
  }

  return NO;
}

//documentation inherited.
- (int) readUnknownJAXBChildElement: (xmlTextReaderPtr) reader
{
  return [super readUnknownJAXBChildElement: reader];
}

//documentation inherited.
- (void) readUnknownJAXBAttribute: (xmlTextReaderPtr) reader
{
  [super readUnknownJAXBAttribute: reader];
}

//documentation inherited.
- (void) writeJAXBAttributes: (xmlTextWriterPtr) writer
{
  int status;

  [super writeJAXBAttributes: writer];

}

//documentation inherited.
- (void) writeJAXBValue: (xmlTextWriterPtr) writer
{
  [super writeJAXBValue: writer];
}

/**
 * Method for writing the child elements.
 *
 * @param writer The writer.
 */
- (void) writeJAXBChildElements: (xmlTextWriterPtr) writer
{
  int status;
  id __item;
  id __item_copy;
  NSEnumerator *__enumerator;

  [super writeJAXBChildElements: writer];

}
@end /* implementation ADMIN_CORENS0SecurityConfiguration (JAXB) */

#endif /* DEF_ADMIN_CORENS0SecurityConfiguration_M */

File: BackEnd/docs/apidocs/admin-core.c
#include <enunciate-common.c>
#ifndef DEF_admin_core_ns0_securityConfiguration_H
#define DEF_admin_core_ns0_securityConfiguration_H

/**
 * (no documentation provided)
 */
struct admin_core_ns0_securityConfiguration {

};

/**
 * Reads a SecurityConfiguration from XML. The reader is assumed to be at the start element.
 *
 * @param reader The XML reader.
 * @return The SecurityConfiguration, or NULL in case of error.
 */
static struct admin_core_ns0_securityConfiguration *xmlTextReaderReadNs0SecurityConfigurationType(xmlTextReaderPtr reader);

/**
 * Writes a SecurityConfiguration to XML.
 *
 * @param writer The XML writer.
 * @param _securityConfiguration The SecurityConfiguration to write.
 * @return The bytes written (may be 0 in case of buffering) or -1 in case of error.
 */
static int xmlTextWriterWriteNs0SecurityConfigurationType(xmlTextWriterPtr writer, struct admin_core_ns0_securityConfiguration *_securityConfiguration);

/**
 * Frees the elements of a SecurityConfiguration.
 *
 * @param _securityConfiguration The SecurityConfiguration to free.
 */
static void freeNs0SecurityConfigurationType(struct admin_core_ns0_securityConfiguration *_securityConfiguration);

#endif /* DEF_admin_core_ns0_securityConfiguration_H */
#ifndef DEF_admin_core_ns0_swaggerResource_H
#define DEF_admin_core_ns0_swaggerResource_H

/**
 * (no documentation provided)
 */
struct admin_core_ns0_swaggerResource {


  /**
   * (no documentation provided)
   */
  xmlChar *swaggerVersion;

  /**
   * (no documentation provided)
   */
  xmlChar *location;

  /**
   * (no documentation provided)
   */
  xmlChar *name;
};

/**
 * Reads a SwaggerResource from XML. The reader is assumed to be at the start element.
 *
 * @param reader The XML reader.
 * @return The SwaggerResource, or NULL in case of error.
 */
static struct admin_core_ns0_swaggerResource *xmlTextReaderReadNs0SwaggerResourceType(xmlTextReaderPtr reader);

/**
 * Writes a SwaggerResource to XML.
 *
 * @param writer The XML writer.
 * @param _swaggerResource The SwaggerResource to write.
 * @return The bytes written (may be 0 in case of buffering) or -1 in case of error.
 */
static int xmlTextWriterWriteNs0SwaggerResourceType(xmlTextWriterPtr writer, struct admin_core_ns0_swaggerResource *_swaggerResource);

/**
 * Frees the elements of a SwaggerResource.
 *
 * @param _swaggerResource The SwaggerResource to free.
 */
static void freeNs0SwaggerResourceType(struct admin_core_ns0_swaggerResource *_swaggerResource);

#endif /* DEF_admin_core_ns0_swaggerResource_H */
#ifndef DEF_admin_core_ns0_uiConfiguration_H
#define DEF_admin_core_ns0_uiConfiguration_H

/**
 * (no documentation provided)
 */
struct admin_core_ns0_uiConfiguration {

};

/**
 * Reads a UiConfiguration from XML. The reader is assumed to be at the start element.
 *
 * @param reader The XML reader.
 * @return The UiConfiguration, or NULL in case of error.
 */
static struct admin_core_ns0_uiConfiguration *xmlTextReaderReadNs0UiConfigurationType(xmlTextReaderPtr reader);

/**
 * Writes a UiConfiguration to XML.
 *
 * @param writer The XML writer.
 * @param _uiConfiguration The UiConfiguration to write.
 * @return The bytes written (may be 0 in case of buffering) or -1 in case of error.
 */
static int xmlTextWriterWriteNs0UiConfigurationType(xmlTextWriterPtr writer, struct admin_core_ns0_uiConfiguration *_uiConfiguration);

/**
 * Frees the elements of a UiConfiguration.
 *
 * @param _uiConfiguration The UiConfiguration to free.
 */
static void freeNs0UiConfigurationType(struct admin_core_ns0_uiConfiguration *_uiConfiguration);

#endif /* DEF_admin_core_ns0_uiConfiguration_H */
#ifndef DEF_admin_core_ns0_securityConfiguration_M
#define DEF_admin_core_ns0_securityConfiguration_M

/**
 * Reads a SecurityConfiguration from XML. The reader is assumed to be at the start element.
 *
 * @return the SecurityConfiguration, or NULL in case of error.
 */
static struct admin_core_ns0_securityConfiguration *xmlTextReaderReadNs0SecurityConfigurationType(xmlTextReaderPtr reader) {
  int status, depth;
  void *_child_accessor;
  struct admin_core_ns0_securityConfiguration *_securityConfiguration = calloc(1, sizeof(struct admin_core_ns0_securityConfiguration));




  return _securityConfiguration;
}

/**
 * Writes a SecurityConfiguration to XML.
 *
 * @param writer The XML writer.
 * @param _securityConfiguration The SecurityConfiguration to write.
 * @return The total bytes written, or -1 on error;
 */
static int xmlTextWriterWriteNs0SecurityConfigurationType(xmlTextWriterPtr writer, struct admin_core_ns0_securityConfiguration *_securityConfiguration) {
  int status, totalBytes = 0, i;
  xmlChar *binaryData;

  return totalBytes;
}

/**
 * Frees the elements of a SecurityConfiguration.
 *
 * @param _securityConfiguration The SecurityConfiguration to free.
 */
static void freeNs0SecurityConfigurationType(struct admin_core_ns0_securityConfiguration *_securityConfiguration) {
  int i;
}
#endif /* DEF_admin_core_ns0_securityConfiguration_M */
#ifndef DEF_admin_core_ns0_swaggerResource_M
#define DEF_admin_core_ns0_swaggerResource_M

/**
 * Reads a SwaggerResource from XML. The reader is assumed to be at the start element.
 *
 * @return the SwaggerResource, or NULL in case of error.
 */
static struct admin_core_ns0_swaggerResource *xmlTextReaderReadNs0SwaggerResourceType(xmlTextReaderPtr reader) {
  int status, depth;
  void *_child_accessor;
  struct admin_core_ns0_swaggerResource *_swaggerResource = calloc(1, sizeof(struct admin_core_ns0_swaggerResource));



  if (xmlTextReaderIsEmptyElement(reader) == 0) {
    depth = xmlTextReaderDepth(reader);//track the depth.
    status = xmlTextReaderAdvanceToNextStartOrEndElement(reader);

    while (xmlTextReaderDepth(reader) > depth) {
      if (status < 1) {
        //panic: XML read error.
#if DEBUG_ENUNCIATE
        printf("Failure to advance to next child element.\n");
#endif
        freeNs0SwaggerResourceType(_swaggerResource);
        free(_swaggerResource);
        return NULL;
      }
      else if (xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT
        && xmlStrcmp(BAD_CAST "swaggerVersion", xmlTextReaderConstLocalName(reader)) == 0
        && xmlTextReaderConstNamespaceUri(reader) == NULL) {

#if DEBUG_ENUNCIATE > 1
        printf("Attempting to read choice {}swaggerVersion of type {http://www.w3.org/2001/XMLSchema}string.\n");
#endif
        _child_accessor = xmlTextReaderReadXsStringType(reader);
        if (_child_accessor == NULL) {
#if DEBUG_ENUNCIATE
          printf("Failed to read choice {}swaggerVersion of type {http://www.w3.org/2001/XMLSchema}string.\n");
#endif
          //panic: unable to read the child element for some reason.
          freeNs0SwaggerResourceType(_swaggerResource);
          free(_swaggerResource);
          return NULL;
        }

        _swaggerResource->swaggerVersion = ((xmlChar*)_child_accessor);
        status = xmlTextReaderAdvanceToNextStartOrEndElement(reader);
      }
      else if (xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT
        && xmlStrcmp(BAD_CAST "location", xmlTextReaderConstLocalName(reader)) == 0
        && xmlTextReaderConstNamespaceUri(reader) == NULL) {

#if DEBUG_ENUNCIATE > 1
        printf("Attempting to read choice {}location of type {http://www.w3.org/2001/XMLSchema}string.\n");
#endif
        _child_accessor = xmlTextReaderReadXsStringType(reader);
        if (_child_accessor == NULL) {
#if DEBUG_ENUNCIATE
          printf("Failed to read choice {}location of type {http://www.w3.org/2001/XMLSchema}string.\n");
#endif
          //panic: unable to read the child element for some reason.
          freeNs0SwaggerResourceType(_swaggerResource);
          free(_swaggerResource);
          return NULL;
        }

        _swaggerResource->location = ((xmlChar*)_child_accessor);
        status = xmlTextReaderAdvanceToNextStartOrEndElement(reader);
      }
      else if (xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT
        && xmlStrcmp(BAD_CAST "name", xmlTextReaderConstLocalName(reader)) == 0
        && xmlTextReaderConstNamespaceUri(reader) == NULL) {

#if DEBUG_ENUNCIATE > 1
        printf("Attempting to read choice {}name of type {http://www.w3.org/2001/XMLSchema}string.\n");
#endif
        _child_accessor = xmlTextReaderReadXsStringType(reader);
        if (_child_accessor == NULL) {
#if DEBUG_ENUNCIATE
          printf("Failed to read choice {}name of type {http://www.w3.org/2001/XMLSchema}string.\n");
#endif
          //panic: unable to read the child element for some reason.
          freeNs0SwaggerResourceType(_swaggerResource);
          free(_swaggerResource);
          return NULL;
        }

        _swaggerResource->name = ((xmlChar*)_child_accessor);
        status = xmlTextReaderAdvanceToNextStartOrEndElement(reader);
      }
      else {
#if DEBUG_ENUNCIATE > 1
        if (xmlTextReaderConstNamespaceUri(reader) == NULL) {
          printf("unknown child element {}%s for type {}swaggerResource.  Skipping...\n",  xmlTextReaderConstLocalName(reader));
        }
        else {
          printf("unknown child element {%s}%s for type {}swaggerResource. Skipping...\n", xmlTextReaderConstNamespaceUri(reader), xmlTextReaderConstLocalName(reader));
        }
#endif
        status = xmlTextReaderSkipElement(reader);
      }
    }
  }

  return _swaggerResource;
}

/**
 * Writes a SwaggerResource to XML.
 *
 * @param writer The XML writer.
 * @param _swaggerResource The SwaggerResource to write.
 * @return The total bytes written, or -1 on error;
 */
static int xmlTextWriterWriteNs0SwaggerResourceType(xmlTextWriterPtr writer, struct admin_core_ns0_swaggerResource *_swaggerResource) {
  int status, totalBytes = 0, i;
  xmlChar *binaryData;
  if (_swaggerResource->swaggerVersion != NULL) {
    status = xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "swaggerVersion", NULL);
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write start element {}swaggerVersion. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;
#if DEBUG_ENUNCIATE > 1
    printf("writing type {http://www.w3.org/2001/XMLSchema}string for element {}swaggerVersion...\n");
#endif
    status = xmlTextWriterWriteXsStringType(writer, (_swaggerResource->swaggerVersion));
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write type {http://www.w3.org/2001/XMLSchema}string for element {}swaggerVersion. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;

    status = xmlTextWriterEndElement(writer);
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write end element {}swaggerVersion. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;
  }
  if (_swaggerResource->location != NULL) {
    status = xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "location", NULL);
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write start element {}location. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;
#if DEBUG_ENUNCIATE > 1
    printf("writing type {http://www.w3.org/2001/XMLSchema}string for element {}location...\n");
#endif
    status = xmlTextWriterWriteXsStringType(writer, (_swaggerResource->location));
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write type {http://www.w3.org/2001/XMLSchema}string for element {}location. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;

    status = xmlTextWriterEndElement(writer);
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write end element {}location. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;
  }
  if (_swaggerResource->name != NULL) {
    status = xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "name", NULL);
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write start element {}name. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;
#if DEBUG_ENUNCIATE > 1
    printf("writing type {http://www.w3.org/2001/XMLSchema}string for element {}name...\n");
#endif
    status = xmlTextWriterWriteXsStringType(writer, (_swaggerResource->name));
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write type {http://www.w3.org/2001/XMLSchema}string for element {}name. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;

    status = xmlTextWriterEndElement(writer);
    if (status < 0) {
#if DEBUG_ENUNCIATE
      printf("Failed to write end element {}name. status: %i\n", status);
#endif
      return status;
    }
    totalBytes += status;
  }

  return totalBytes;
}

/**
 * Frees the elements of a SwaggerResource.
 *
 * @param _swaggerResource The SwaggerResource to free.
 */
static void freeNs0SwaggerResourceType(struct admin_core_ns0_swaggerResource *_swaggerResource) {
  int i;
  if (_swaggerResource->swaggerVersion != NULL) {
#if DEBUG_ENUNCIATE > 1
    printf("Freeing type of accessor swaggerVersion of type admin_core_ns0_swaggerResource...\n");
#endif
    freeXsStringType(_swaggerResource->swaggerVersion);
#if DEBUG_ENUNCIATE > 1
    printf("Freeing accessor swaggerVersion of type admin_core_ns0_swaggerResource...\n");
#endif
    free(_swaggerResource->swaggerVersion);
  }
  if (_swaggerResource->location != NULL) {
#if DEBUG_ENUNCIATE > 1
    printf("Freeing type of accessor location of type admin_core_ns0_swaggerResource...\n");
#endif
    freeXsStringType(_swaggerResource->location);
#if DEBUG_ENUNCIATE > 1
    printf("Freeing accessor location of type admin_core_ns0_swaggerResource...\n");
#endif
    free(_swaggerResource->location);
  }
  if (_swaggerResource->name != NULL) {
#if DEBUG_ENUNCIATE > 1
    printf("Freeing type of accessor name of type admin_core_ns0_swaggerResource...\n");
#endif
    freeXsStringType(_swaggerResource->name);
#if DEBUG_ENUNCIATE > 1
    printf("Freeing accessor name of type admin_core_ns0_swaggerResource...\n");
#endif
    free(_swaggerResource->name);
  }
}
#endif /* DEF_admin_core_ns0_swaggerResource_M */
#ifndef DEF_admin_core_ns0_uiConfiguration_M
#define DEF_admin_core_ns0_uiConfiguration_M

/**
 * Reads a UiConfiguration from XML. The reader is assumed to be at the start element.
 *
 * @return the UiConfiguration, or NULL in case of error.
 */
static struct admin_core_ns0_uiConfiguration *xmlTextReaderReadNs0UiConfigurationType(xmlTextReaderPtr reader) {
  int status, depth;
  void *_child_accessor;
  struct admin_core_ns0_uiConfiguration *_uiConfiguration = calloc(1, sizeof(struct admin_core_ns0_uiConfiguration));




  return _uiConfiguration;
}

/**
 * Writes a UiConfiguration to XML.
 *
 * @param writer The XML writer.
 * @param _uiConfiguration The UiConfiguration to write.
 * @return The total bytes written, or -1 on error;
 */
static int xmlTextWriterWriteNs0UiConfigurationType(xmlTextWriterPtr writer, struct admin_core_ns0_uiConfiguration *_uiConfiguration) {
  int status, totalBytes = 0, i;
  xmlChar *binaryData;

  return totalBytes;
}

/**
 * Frees the elements of a UiConfiguration.
 *
 * @param _uiConfiguration The UiConfiguration to free.
 */
static void freeNs0UiConfigurationType(struct admin_core_ns0_uiConfiguration *_uiConfiguration) {
  int i;
}
#endif /* DEF_admin_core_ns0_uiConfiguration_M */
